### DISCUSSION TITLE
所有 dbus Name 都变成 activatable?

### DISCUSSION THREAD

**Discussion by cheese** [fcitx/fcitx5#479] [Ideas]
以 Flatpak 打包的 Fcitx5 为例，现在只有 `org.fcitx.Fcitx5` 是 activatable 的，能否将其它的 Name （比如 `org.fcitx.Fcitx-0`、 `org.fcitx.Fcitx-99`、`org.freedesktop.portal.Fcitx`）全部都变成 activatable？那就不需要手动设置 Fcitx5 服务自动启动。

**Comment by wengxt** [comment]
不能。首先前两个都是根据 display name猜测的，并不是稳定的名称。
另外im module 并没有直接调用对应 service 的名称。可以避免一旦发生crash loop导致的问题。

**Reply by wengxt** [reply]
一些特定的 setup 更需要 compositor 启动输入法，输入法不能自己随意启动，否则无法获得对应的权限

**Comment by tinywrkb** [comment]
For Flatpak, using the [Background portal](https://flatpak.github.io/xdg-desktop-portal/#gdbus-org.freedesktop.impl.portal.Background) (see `EnableAutostart` method) might be more fitting for auto-starting the service. Don't expect support in anything but Gnome and KDE.
